File: simulation/build-array-from-permutation.py
from typing import List

class Solution:
    def buildArray(self, nums: List[int]) -> List[int]:
        # Initialize an empty list to store the result
        ans = []

        # Iterate over each index in the input list
        for i in range(len(nums)):
            # For each index i, append nums[nums[i]] to the result list
            value = nums[nums[i]]  
            ans.append(value)

        return ans

        

File: simulation/build-array-from-permutation.java
import java.util.Arrays;

public class Solution {
    public static void main(String[] args) {
        Solution sol = new Solution();

    }

    public int[] buildArray(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];

        for (int i = 0; i < n; i++) {
            ans[i] = nums[nums[i]];
        }

        return ans;
    }
}